src/shape/2d_primitives.js

```diff
@@ -444,33 +444,85 @@ define(function (require) {
   * every angle at ninety degrees. By default, the first two parameters set
   * the location of the upper-left corner, the third sets the width, and the
   * fourth sets the height. The way these parameters are interpreted, however,
-  * may be changed with the rectMode() function.
+  * may be changed with the rectMode() function. If provided, the fifth, sixth
+  * seventh and eighth parameters, if specified, determine corner radius for
+  * the top-right, top-left, lower-right and lower-left corners, respectively.
+  * An omitted corner radius parameter is set to the value of the previously
+  * specified radius value in the parameter list.
   *
   * @method rect
-  * @param  {Number} a x-coordinate of the rectangle
-  * @param  {Number} b y-coordinate of the rectangle
-  * @param  {Number} c width of the rectangle
-  * @param  {Number} d height of the rectangle
-  * @return {p5}       the p5 object
+  * @param  {Number} x  x-coordinate of the rectangle.
+  * @param  {Number} y  y-coordinate of the rectangle.
+  * @param  {Number} w  width of the rectangle.
+  * @param  {Number} h  height of the rectangle.
+  * @param  {Number} [tl] optional radius of top-left corner.
+  * @param  {Number} [tr] optional radius of top-right corner.
+  * @param  {Number} [br] optional radius of bottom-right corner.
+  * @param  {Number} [bl] optional radius of bottom-left corner.
+  * @return {p5}          the p5 object.
   * @example
   * <div>
   * <code>
+  * // Draw a rectangle at location (30, 25) with a width and height of 55.
   * rect(30, 20, 55, 55);
+  * // Draw a rectangle with rounded corners, each having a radius of 20.
+  * rect(30, 20, 55, 55, 20);
+  * // Draw a rectangle with rounded corners having the following radii:
+  * // top-left = 20, top-right = 15, bottom-right = 10, bottom-left = 5.
+  * rect(30, 20, 55, 55, 20, 15, 10, 5)
   * </code>
   * </div>
```

Comment:
Because of how the documentation is generated, each example should be separated into its own code block so that it renders as its own sketch in the docs. Just add a
```js
 * </code>
 * </div>
 *
 * <div>
 * <code>
```
between each example (so between lines 467 & 468, and lines 469 & 470, in this case)